src/main/java/com/aws/greengrass/clientdevices/auth/iot/Certificate.java

```diff
@@ -145,9 +145,20 @@ public static void updateMetadataTrustDurationMinutes(int newTrustDuration) {
         metadataTrustDurationMinutes.set(newTrustDuration);
     }
 
+    /**
+     * Following opt-in offline auth, verifies the certificate status trust.
+     * Metadata trust verification is enabled when configured trust duration is non-zero.
+     * Trust verification is disabled by default.
+     *
+     * @return whether the status is verified within configured trust duration
+     */
     private boolean isStatusTrusted() {
```

Comment:
It would be nice to move this into an interface that both of those object implement kind of 

```java
interface Expirable {
      default boolean hasExpired(int maxMinutesToLive) {
        if (maxMinutesToLive > 0) {
            Instant validTill = statusLastUpdated.plus(maxMinutesToLive, ChronoUnit.MINUTES);
            return validTill.isAfter(Instant.now());
        }
        return true;
      }
}
```

and then have 

```java
    public Status getStatus() {
        if (hasExpired(metadataTrustDurationMinutes.get())) {
            return status;
        }
        return Status.UNKNOWN;
    }
```

```java
class Certificate implements Expirable
```

Reply:
Good call. Added a common `Trustable` interface. Thanks